core/src/main/java/org/fao/geonet/util/XslUtil.java

```diff
@@ -827,44 +852,27 @@ String twoCharLangCode(String iso3LangCode, String defaultValue) {
             if (defaultValue != null) {
                 return defaultValue;
             } else {
-                return twoCharLangCode(Geonet.DEFAULT_LANGUAGE);
-            }
-        } else {
-            if (iso3LangCode.equalsIgnoreCase("FRA")) {
-                return "FR";
+                iso3LangCode = Geonet.DEFAULT_LANGUAGE;
             }
+        }
+        String iso2LangCode = null;
 
-            if (iso3LangCode.equalsIgnoreCase("DEU")) {
-                return "DE";
-            }
-            String iso2LangCode = null;
+        // Catch language entries longer than 3 characters with a semicolon
+        if (iso3LangCode.length() > 3 && (iso3LangCode.indexOf(';') != -1)) {
+            iso3LangCode = iso3LangCode.substring(0, 3);
```

Comment:
Is it possible the input as fr;CAN. Then the substring will endup getting "fr;" instead? If so, string split(";")[1] may be safer. After that, will probably check if it's 3 characters or not.